GX-TimeFrequency/utilities/gx_tabulate_minimax.F90

```diff
@@ -0,0 +1,250 @@
+    real(dp), parameter :: one = 1._dp
+    !scalars
+    integer :: ii, it, iw, ierr, num_integ_points, unt, py_unt, nargs, iostat, ntau = -1
+    real(dp) :: emin = huge(one), emax = huge(one), eratio, cosft_duality_error
+    character(len=1024) :: msg, arg, arg_val, iomsg, command
+    logical :: do_plot = .False.
+    !arrays
+    real(dp) :: max_errors(3)
+    real(dp), allocatable :: tau_mesh(:), tau_wgs(:), iw_mesh(:), iw_wgs(:)
+    real(dp), allocatable :: cosft_wt(:,:), cosft_tw(:,:), sinft_wt(:,:)
+
+    ! **************************************************************************************************
+
+    nargs = command_argument_count()
+
+    if (nargs < 1) then
+        call print_help()
+        stop
+    end if
+
+    call get_command_argument(1, command)
+
+    do ii=2,nargs
+        call get_command_argument(ii, arg)
+        if (arg(1:1) /= "-") cycle
+
+        select case (arg)
+        case ("-emin")
+            call get_command_argument(ii + 1, arg_val)
+            read(arg_val, *, iostat=iostat, iomsg=iomsg) emin
+            call iocheck()
+
+        case ("-emax")
+            call get_command_argument(ii + 1, arg_val)
+            read(arg_val, *, iostat=iostat, iomsg=iomsg) emax
+            call iocheck()
+
+        case ("-ntau")
+            call get_command_argument(ii + 1, arg_val)
+            read(arg_val, *, iostat=iostat, iomsg=iomsg) ntau
+            call iocheck()
+
+        case ("-plot", "-p")
+            do_plot = .True.
+
+        case ("-h", "--help")
+            write(std_out, *)"Help me!"
+            stop 0
+
+        case default
+            write(std_out, *)"Invalid argument:", trim(arg)
+            stop 1
+        end select
+    end do
+
+    if (emin == huge(one) .or. emax == huge(one)) then
+        write(std_out,*)"emin and emax arguments are mandatory"; stop 1
+    end if
+
+    if (emin <= 1e-6) then
+        write(std_out,*)"emin should be greater than zero"; stop 1
+    end if
+
+    eratio = emax / emin
+
+    select case (command)
+    case ("table")
+
+        call my_open("_gx_minimax_table.csv", unt)
+        write(msg, "(a)") &
+                "num_points ierr cosft_duality_error max_err_costf_t_to_w max_err_costf_w_to_t max_err_sintf_t_to_w eratio"
+        call dumps(msg, [std_out, unt])
+
+        do ii=1, size(tau_npoints_supported)
+            num_integ_points = tau_npoints_supported(ii)
+
+            call gx_minimax_grid(num_integ_points, emin, emax, &
+                    tau_mesh, tau_wgs, iw_mesh, iw_wgs, &
+                    cosft_wt, cosft_tw, sinft_wt, &
+                    max_errors, cosft_duality_error, ierr)
+
+            write(msg, "(2(i0,2x), *(es16.8,2x))") num_integ_points, ierr, cosft_duality_error, max_errors, eratio
+            call dumps(msg, [std_out, unt])
+        end do
+        close(unt)
+
+        ! Write python script to plot the data.
+        open(file="_gx_minimax_table.py", newunit=py_unt, action="write")
```

Comment:
Comment: one of the things that I hate the most from Fortran is its limited interop capabilities, forcing one to write stuff like this

Reply:
In this context, I think this is unnecessary - one could just provide the python script separately. However this is Matteo's design decision and I am not altering his design decision or implementation.  

In general, one can use ctypes to write python bindings for calling minimax. I just don't know how to do it when one has allocatable arrays with memory assigned in the external library. I think Matteo may change the memory management to faciliate.